mountpoint-s3/src/fuse/session.rs

```diff
@@ -89,8 +92,29 @@ impl FuseSession {
     /// Block until the file system is unmounted or this process is interrupted via SIGTERM/SIGINT.
     /// When that happens, unmount the file system (if it hasn't been already unmounted).
     pub fn join(mut self) -> anyhow::Result<()> {
-        let msg = self.receiver.recv();
-        trace!("received message {msg:?}, closing filesystem session");
+        let mut sys =
+            System::new_with_specifics(RefreshKind::new().with_processes(ProcessRefreshKind::new().with_memory()));
+        let pid = Pid::from(process::id() as usize);
+        let reporting_period = Duration::from_secs(5);
+        loop {
+            match self.receiver.recv_timeout(reporting_period) {
+                Ok(msg) => {
+                    trace!("received message {msg:?}, closing filesystem session");
+                    break;
+                }
+                Err(RecvTimeoutError::Disconnected) => {
+                    trace!("fuse session's channel is disconnected, closing filesystem session");
+                    break;
+                }
+                Err(RecvTimeoutError::Timeout) => {
+                    // Report process level stats
+                    sys.refresh_process_specifics(pid, ProcessRefreshKind::new().with_memory());
+                    if let Some(process) = sys.process(pid) {
+                        metrics::gauge!("mountpoint_s3.total_memory_usage").set(process.memory() as f64);
```

Comment:
Cool, we'll have a memory usage reported!

What bothers me is the time when we record this value: it is not synchronised with the recording times of other memory metrics,  e.g. `prefetch.bytes_in_queue`. This `total_memory_usage` will likely be queried from procfs in the beginning of [the aggregation period](https://github.com/awslabs/mountpoint-s3/blob/main/mountpoint-s3/src/metrics.rs#L22), but published with others at the end of the period.

I am suggesting to query `process.memory()` just before [publishing](https://github.com/awslabs/mountpoint-s3/blob/main/mountpoint-s3/src/metrics.rs#L49) the metrics. Though we must be sure that it does not delay the publishing significantly. Does`refresh_process_specifics` takes nanoseconds or higher order of magnitude?

Reply:
Good point. I was thinking about putting the `total_memory_usage` in the metric sink itself too, but wasn't really sure. But this might be a good reason.

The time it takes for refreshing should be in nanoseconds, we only have to make sure not to initialise the `System` struct every time.